cis194/week2/daniel-deng/src/CIS194/Homework02/Exercise01.hs

```diff
@@ -0,0 +1,48 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module CIS194.Homework02.Exercise01 (parseMessage, parseMessages) where
+
+import qualified CIS194.Homework02.Log as Log
+
+import           Data.Text             hiding (lines, map)
+import           Text.Parsec
+import           Text.Parsec.Text
+
+skipSpaces :: Parser ()
+skipSpaces = skipMany1 $ char ' '
+
+skipSpacesIfAny :: Parser ()
+skipSpacesIfAny = skipMany $ char ' '
+
+digitsAfter :: Char -> Parser String
+digitsAfter c = char c *> skipSpaces *> many1 digit
+
+(~>) :: Char -> value -> Parser value
```

Comment:
Usually we put `a`, `b`, `c` ..... as name of polymorphic type. Because they are arbitrary so `names don't matter maters ` here.
```suggestion
(~>) :: Char -> a -> Parser a
```